pkg/cloudevents/transport/http/transport.go

```diff
@@ -171,26 +176,28 @@ func (t *Transport) StartReceiver(ctx context.Context) error {
 		}
 		t.realPort = listener.Addr().(*net.TCPAddr).Port
 
-		go func() {
-			log.Print(t.server.Serve(listener))
-			t.server = nil
+		// Shutdown
+		defer func() {
 			t.realPort = 0
+			t.server.Close()
```

Comment:
Might be better to use graceful `Shutdown(ctx)` here, but I'm not sure how to choose a shutdown timeout.